include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,873 @@
+/* Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:	BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP Compression
+ */
+
+#ifndef ODP_API_COMP_H_
+#define ODP_API_COMP_H_
+
+#include <odp/visibility_begin.h>
+#include <odp/api/support.h>
+#include <odp/api/packet.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup odp_compression ODP COMP
+ *  ODP Compression is an API set to do compression+hash or decompression+hash
+ *  operations on data. Hash is calculated on plaintext.
+ *
+ *  if opcode = ODP_COMP_COMPRESS, then it will apply hash and then compress,
+ *  if opcode = ODP_COMP_DECOMPRESS, then it will decompress and then apply
+ *  hash.
```

Comment:
This is description of how de/-compression operation work. This should be specified at operation documentation, and optimally only there. Otherwise, we are in danger of this spec getting deprecated (== in conflict with operation documentation) without noticing.

Reply:
" This should be specified at operation documentation" means which place? user guide Or ?

Do you mean I should remove this description from here? Or 
"+ *  if opcode = ODP_COMP_COMPRESS, then it will apply hash and then compress,
 + *  if opcode = ODP_COMP_DECOMPRESS, then it will decompress and then apply
 + *  hash."

Reply:
E.g. move to odp_comp_op() documentation (in this file) and then refer to that from other comp operation functions.